### PR TITLE
feat(testcontainer): optimize test container

### PR DESCRIPTION
**Pull request by AlexStocks**
### Purpose or design rationale of this PR

*Describe your change. Make sure to answer these three questions: What does this PR do? Why does it do it? How does it do it?*

optimize docker compose

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: common/testcontainers/testcontainers.go [hunk 1]
```diff
@@ -20,9 +25,11 @@ type TestcontainerApps struct {
 	postgresContainer *postgres.PostgresContainer
 	l1GethContainer   *testcontainers.DockerContainer
 	l2GethContainer   *testcontainers.DockerContainer
+	poSL1Container    compose.ComposeStack
```
Comment (georgehao) [inline common/testcontainers/testcontainers.go:28]: @colinlyguo how about change to `posL1Container`

File: Makefile [hunk 2]
```diff
@@ -55,3 +55,4 @@ run_test_docker: ## run Docker image for local testing on M1/M2 Silicon Mac
 
 clean: ## Empty out the bin folder
 	@rm -rf build/bin
+
```
Comment (georgehao) [inline Makefile:58]: remove this

File: build/.golangci.yml [hunk 3]
```diff
@@ -213,7 +213,7 @@ issues:
       linters:
         - errcheck
         - gosec
-        
+
```
Comment (georgehao) [inline build/.golangci.yml]: ditto

File: rollup/internal/orm/orm_test.go [hunk 4]
```diff
@@ -12,13 +12,12 @@ import (
 	"github.com/stretchr/testify/assert"
 	"gorm.io/gorm"
 
-	"scroll-tech/database/migrate"
-
```
Comment (georgehao) [inline rollup/internal/orm/orm_test.go:16]: recover this

File: common/testcontainers/testcontainers.go [hunk 5]
```diff
@@ -106,6 +113,69 @@ func (t *TestcontainerApps) StartL2GethContainer() error {
...
+	if rootDir, err = findProjectRootDir(); err != nil {
+		return fmt.Errorf("failed to find project root directory: %v", err)
+	}
+
+	if rnd, err = rand.Int(rand.Reader, big.NewInt(65536-1024)); err != nil {
+		return fmt.Errorf("failed to generate a random: %v", err)
+	}
+	t.poSL1GethPort = int64(int(rnd.Int64()) + 1024)
+
+	if err = os.Setenv("GETH_HTTP_PORT", fmt.Sprintf("%d", t.poSL1GethPort)); err != nil {
```
Comment (colinlyguo) [inline common/testcontainers/testcontainers.go:137]: Could you help remove these env settings? No need for them now.

File: common/testcontainers/testcontainers.go [hunk 6]
```diff
@@ -106,6 +113,69 @@ func (t *TestcontainerApps) StartL2GethContainer() error {
...
+		rnd               *big.Int
+		dockerComposeFile string
+		env               = map[string]string{}
+	)
+
+	if rootDir, err = findProjectRootDir(); err != nil {
+		return fmt.Errorf("failed to find project root directory: %v", err)
+	}
+
+	if rnd, err = rand.Int(rand.Reader, big.NewInt(65536-1024)); err != nil {
```
Comment (colinlyguo) [inline common/testcontainers/testcontainers.go:132]: Can also try to make this port static.

File: Makefile (hunk 2 above)
Comment (AlexStocks) [reply Makefile:58]: done

**Pull request closed without merging**